velox/functions/lib/window/tests/WindowTestBase.cpp

```diff
@@ -48,6 +49,79 @@ WindowTestBase::QueryInfo WindowTestBase::buildWindowQuery(
   return {op, functionSql, querySql};
 }
 
+namespace {
+std::vector<std::string> commaSplit(const std::string& input) {
+  std::istringstream tokenStream(input);
+  std::string token;
+  std::vector<std::string> splits;
+  while (std::getline(tokenStream, token, ',')) {
+    splits.push_back(token);
+  }
+
+  return splits;
+}
+}; // namespace
+
+WindowTestBase::QueryInfo WindowTestBase::buildStreamingWindowQuery(
+    const std::vector<RowVectorPtr>& input,
+    const std::string& function,
+    const std::string& overClause,
+    const std::string& frameClause) {
+  std::string functionSql =
+      fmt::format("{} over ({} {})", function, overClause, frameClause);
+  core::PlanNodePtr op = nullptr;
+  std::vector<std::string> orderByClauses;
+
+  auto partitionByIndex = overClause.find(kPartitionByStr);
+  auto orderByIndex = overClause.find(kOrderByStr);
+
+  // Extract the partition by keys.
```

Comment:
Hmm... would it be possible to remove the need for this parsing logic?

Reply:
@mbasmanova Before testing the streaming window, it is necessary for us to sort the input data based on the partition key and order by key. Therefore, this portion of logic is required

Reply:
@JkSelf I was referring to the logic of parsing overClause string. I wonder if it would be possible to refactor the test so that this function receives all the necessary information in a convenient format and doens't need to parse overClause SQL. In case this is not possible, let's use DuckParser for parsing.

Reply:
@mbasmanova Updated this logic to use DuckParser for parsing.